doc/manual/src/glossary.md

```diff
@@ -208,44 +208,85 @@
+  - [host platform]{#gloss-host-platform}
 
-    When discussing a build that does not produce a compiler, we only use [build system] and [host system] to describe it; we try to avoid discussing the actual build from the perspective its compiler's configuration.
+    [host platform]: #gloss-host-platform
 
-    This is based on the [GCC system terminology](#gloss-gcc-system-terminology).
+    The [platform] where build products will run on, typically a dash-separated string.
 
-    If the output of a compiler is to be run in a [derivation], the compiler's [target system] must be executable by the [derivation system](language/derivations.md#attr-system).
+    Programs that run natively in a [derivation] must have a [build configuration] with a [host platform] that can by executed by the [derivation system](language/derivations.md#attr-system).
+
+    This term is based on the [GCC system terminology](#gloss-gcc-system-terminology).
+
+  - [target platform]{#gloss-target-platform}
+
+    [target platform]: #gloss-target-platform
+
+    The [platform] for which a compiler will generate code, typically a dash-separated string.
+
+    When discussing a build that does not produce a compiler, we only use [build platform] and [host platform] to describe it. This also applies to compilers with a dynamic target, which is not part of the [build configuration].
+
+    We try to avoid discussing the actual build from the perspective of its compiler, instead focusing on the build itself.
+    Otherwise we won't have a term to distinguish the actual [target platform] from what we should have called the [host platform].
+
+    If the output of a compiler is to be run in a [derivation], the compiler's [target platform] must be executable by the [derivation system](language/derivations.md#attr-system).
+
+    This term is based on the [GCC system terminology](#gloss-gcc-system-terminology).
 
   - [GCC system terminology]{#gloss-gcc-system-terminology}
 
     The Nix ecosystem uses the [GCC system terminology](https://gcc.gnu.org/onlinedocs/gccint/Configure-Terms.html) for the names and roles of _build_, _host_, and _target_, but may apply the term _cross_ more loosely.
 
-    See [host system], [build system] and [target system] for how each relates to [derivation].
+    See [host platform], [build platform] and [target platform] for how each relates to [derivation].
+
+  - [build configuration]{#gloss-build-configuration}
+
+    [build configuration]: #gloss-build-configuration
+
+    The software configuration whose effects enable the build process and most of which also influence the [build product](#gloss-build-product).
+
+    Build configuration does not have a single agreed upon definition, but consists of a potentially broad set of settings that include:
+    - [build platform], often implicitly
+    - [host platform]
+    - [target platform], only when building a compiler with a static target platform
```

Comment:
```suggestion
    - [target platform], only for compilers where their compilation target platform is determined when the compiler itself is build
```

This is a bit lengthy and clumsy, but  think we should make that unambiguous. Maybe you have a more elegant way of saying this without obscuring the meaning.